compiler/tests/testSuite.bal

```diff
@@ -57,6 +58,40 @@ function testCompileEU(string path, string kind) returns file:Error|io:Error? {
     }
 }
 
+function verifyErrorFilename(err:Detail detail, string path) returns string {
+    test:assertTrue(detail.location is err:Location, "error without location");
+    string filename =(<err:Location>detail.location).filename;
+    test:assertNotEquals(filename.length(), 0, "error with an empty filename");
+    string? importName = checkpanic importFilename(path, filename);
+    if importName is string {
+        test:assertEquals(filename, checkpanic file:basename(importName), "invalid error filename");
+        return importName;
+    }
+    else {
+        test:assertEquals(filename, path, "invalid error filename");
+        return path;
+    }
+}
+
+function importFilename(string path, string errFilename) returns string|file:Error? {
```

Comment:
Current name doesn't give a good idea about what it is doing.
How about something like:
```suggestion
function searchSubModuleFile(string rootModule, string filename) returns string|file:Error? {
```

I personally think it's better if the error trace had the sub-module name, then we wouldn't need this function. Let's discuss that option separately.

Reply:
function removed. All errors should show the path (either absolute or relative) of the file that caused the error